test/util/listener/listener.go

```diff
@@ -36,10 +38,14 @@ func (l *Listener) Accept() (net.Conn, error) {
 }
 
 func (l *Listener) Close() error {
+	l.Lock()
+	defer l.Unlock()
+
 	if !l.closed {
 		close(l.c)
 		l.closed = true
 	}
+
 	return nil
```

Comment:
So we can do this way:
```suggestion
	l.closeOnce.Do(func() {
	    close(l.c)
	    l.closed = true
	})
	return nil
```

Reply:
Good idea. Tks!